orbit/pkg/update/nudge.go

```diff
@@ -43,52 +69,147 @@ func (n *NudgeConfigFetcher) GetConfig() (*fleet.OrbitConfig, error) {
 		return nil, nil
 	}
 
-	var foundTarget bool
-	for _, t := range n.UpdateRunner.opt.Targets {
-		if t == "nudge" {
-			foundTarget = true
-		}
+	if cfg.NudgeConfig == nil {
+		log.Info().Msg("empty nudge config, removing nudge as target")
+		// TODO(roberto): by early returning and removing the target from the
+		// runner/updater we ensure Nudge won't be opened/updated again
+		// but we don't actually remove the file from disk. We
+		// knowingly decided to do this as a post MVP optimization.
+		n.opt.UpdateRunner.RemoveRunnerOptTarget("nudge")
+		n.opt.UpdateRunner.updater.RemoveTargetInfo("nudge")
+		return cfg, nil
+	}
+
+	updaterHasTarget := n.opt.UpdateRunner.HasRunnerOptTarget("nudge")
+	runnerHasLocalHash := n.opt.UpdateRunner.HasLocalHash("nudge")
+	if !updaterHasTarget || !runnerHasLocalHash {
+		log.Info().Msg("refreshing the update runner config with Nudge targets and hashes")
+		return cfg, n.setTargetsAndHashes()
+	}
+	if err := n.opt.UpdateRunner.StoreLocalHash("nudge"); err != nil {
+		return cfg, err
+	}
+
+	if err := n.configure(*cfg.NudgeConfig); err != nil {
+		log.Info().Err(err).Msg("nudge configuration")
+		return cfg, err
 	}
 
-	switch {
-	case !foundTarget && cfg.NudgeConfig != nil:
-		addNudgeTarget(n.UpdateRunner)
-	case foundTarget && cfg.NudgeConfig == nil:
-		removeNudgeTarget(n.UpdateRunner)
-	default:
-		// ok
+	if err := n.launch(); err != nil {
+		log.Info().Err(err).Msg("nudge launch")
+		return cfg, err
 	}
 
 	return cfg, nil
 }
 
-func addNudgeTarget(r *Runner) {
-	log.Info().Msg("adding nudge as target")
-	r.AddRunnerOptTarget("nudge")
-	r.updater.SetTargetInfo(
-		"nudge",
-		"macos",
-		"stable",
-		"nudge.app.tar.gz",
-		[]string{"Nudge.app", "Contents", "MacOS", "Nudge"},
-	)
+func (n *NudgeConfigFetcher) setTargetsAndHashes() error {
+	n.opt.UpdateRunner.AddRunnerOptTarget("nudge")
+	n.opt.UpdateRunner.updater.SetTargetInfo("nudge", NudgeMacOSTarget)
+	// we don't want to keep nudge as a target if we failed to update the
+	// cached hashes in the runner.
+	if err := n.opt.UpdateRunner.StoreLocalHash("nudge"); err != nil {
+		n.opt.UpdateRunner.RemoveRunnerOptTarget("nudge")
+		n.opt.UpdateRunner.updater.RemoveTargetInfo("nudge")
+		return err
+	}
+	return nil
 }
 
-func removeNudgeTarget(r *Runner) {
-	log.Info().Msg("removing nudge as target")
-	r.RemoveRunnerOptTarget("nudge")
-	r.updater.RemoveTargetInfo("nudge")
+func (n *NudgeConfigFetcher) configure(nudgeCfg fleet.NudgeConfig) error {
+	jsonCfg, err := json.Marshal(nudgeCfg)
+	if err != nil {
+		return err
+	}
+
+	cfgFile := filepath.Join(n.opt.RootDir, nudgeConfigFile)
+	writeConfig := func() error {
+		return os.WriteFile(cfgFile, jsonCfg, constant.DefaultWorldReadableFileMode)
```

Comment:
Just double-checking that this file needs permissions at the world level?

Reply:
I think you're right, thanks! To be sure, I tested with `DefaultFileMode` and didn't find any problems.